### PR TITLE
Network interface metrics

### LINKED ISSUE #330: Provide support for scraping metrics for Azure Virtual Network
Provide support for scraping metrics for Azure Virtual Network

### PR DESCRIPTION
**Pull request by rguthriemsft**
Fixes #330 

## Proposed Changes
Provides support for scraping Azure Network Interfaces.  

Example configuration:

```YAML
  - name: demo_azuresnetworkinterface_bytesreceivedrate
    description: "Number of bytes the Network Interface sent"
    resourceType: NetworkInterface
    networkInterfaceName: promitor-network-interface
    azureMetricConfiguration:
      metricName: BytesReceivedRate
      aggregation:
        type: Average
```

### CODE REVIEW

File: docs/configuration/metrics/network-interfaces.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+layout: default
+title: Azure Network Interface Declaration
+---
+
+## Azure Network Interface
```
Comment (tomkerkhove) [inline docs/configuration/metrics/network-interfaces.md]: Thought - Should we elaborate on network interface vs network? For me this is not clear as a non-infra person but I think they are the same or not?

**Comment by tomkerkhove** [comment]
@rguthriemsft 	The PR looks fine for me, thanks! Maybe good to quickly discuss my comment with @jsturtevant 	& @brusMX.

For the rest we can merge I think.

File: docs/configuration/metrics/network-interfaces.md (hunk 1 above)
Comment (rguthriemsft) [reply docs/configuration/metrics/network-interfaces.md]: I added a link to Microsoft docs for network interface.

**Comment by brusMX** [comment]
@rguthriemsft  Can you fix the conflicting file of .gitignore?

File: src/Promitor.Scraper.Tests.Unit/Serialization/MetricsDeclaration/MetricsDeclarationWithNetworkInterfaceYamlSerializationTests.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+namespace Promitor.Scraper.Tests.Unit.Serialization.MetricsDeclaration
+{
+    [Category(category: "Unit")]
+    public class MetricsDeclarationWithNetworkInterfaceYamlSerializationTests : YamlSerializationTests<NetworkInterfaceMetricDefinition>
+    {
+        [Theory]
+        [InlineData("promitor1")]
+        [InlineData(data: null)]
+        public void YamlSerialization_SerializeAndDeserializeValidConfigForNetworkInterface_SucceedsWithIdenticalOutput(string resourceGroupName)
```
Comment (brandonh-msft) [inline src/Promitor.Scraper.Tests.Unit/Serialization/MetricsDeclaration/MetricsDeclarationWithNetworkInterfaceYamlSerializationTests.cs]: If you have a look at the latest tests, there are more parameters for them due to the metric scraping interval changes that have been merged.

**Pull request merged**